### ISSUE TITLE
Plugin does not build with strict includes

### ISSUE THREAD

**Issue by knapsu** [getnamo/SocketIOClient-Unreal#419]
Plugin fails to build when strict include checks are enabled. This can be easily tested with UAT command
```
.\RunUAT.bat BuildPlugin -Plugin="{workspacePath}\Engine\Plugins\SocketIOClient\SocketIOClient.uplugin" -StrictIncludes -Package="{workspacePath}\LocalBuilds"
```
What `-StrictIncludes` does is adding `-DisableUnity` `-NoPCH` `-NoSharedPCH` to UBT when compiling. This ensures that each code file has all the needed includes and therefore is of a good quality on that side.

Will create a pull request that fixes some of the issues.

**Comment by knapsu** [comment]
Pull request created. This is a partial fix because there is still a thing with `sio_socket.h` that I was unable to solve.

**Comment by getnamo** [comment]
Good idea, thanks for getting the PR done; merged the includes update.
